src/sdl2_compat.c

```diff
@@ -3914,6 +3914,231 @@ DECLSPEC void SDLCALL SDL_SIMDFree(void *ptr)
 }
 
 
+
+static SDL_bool SDL_IsSupportedAudioFormat(const SDL_AudioFormat fmt)
+{
+    switch (fmt) {
+    case AUDIO_U8:
+    case AUDIO_S8:
+    case AUDIO_U16LSB:
+    case AUDIO_S16LSB:
+    case AUDIO_U16MSB:
+    case AUDIO_S16MSB:
+    case AUDIO_S32LSB:
+    case AUDIO_S32MSB:
+    case AUDIO_F32LSB:
+    case AUDIO_F32MSB:
+        return SDL_TRUE; /* supported. */
+
+    default:
+        break;
+    }
+
+    return SDL_FALSE; /* unsupported. */
+}
+
+static SDL_bool SDL_IsSupportedChannelCount(const int channels)
+{
+    return ((channels >= 1) && (channels <= 8)) ? SDL_TRUE : SDL_FALSE;
+}
+
+
+typedef struct {
+    SDL_AudioFormat src_format;
+    Uint8 src_channels;
+    int src_rate;
+    SDL_AudioFormat dst_format;
+    Uint8 dst_channels;
+    int dst_rate;
+} AudioParam;
+
+
+DECLSPEC int SDLCALL SDL_BuildAudioCVT(SDL_AudioCVT *cvt,
+                                       SDL_AudioFormat src_format,
+                                       Uint8 src_channels,
+                                       int src_rate,
+                                       SDL_AudioFormat dst_format,
+                                       Uint8 dst_channels,
+                                       int dst_rate)
+{
+    SDL_AudioStream *stream;
+
+    /* Sanity check target pointer */
+    if (cvt == NULL) {
+        return SDL_InvalidParamError("cvt");
+    }
+
+    /* Make sure we zero out the audio conversion before error checking */
+    SDL_zerop(cvt);
+
+#if 0
+    if (!SDL_IsSupportedAudioFormat(src_format)) {
+        return SDL_SetError("Invalid source format");
+    }
+    if (!SDL_IsSupportedAudioFormat(dst_format)) {
+        return SDL_SetError("Invalid destination format");
+    }
+    if (!SDL_IsSupportedChannelCount(src_channels)) {
+        return SDL_SetError("Invalid source channels");
+    }
+    if (!SDL_IsSupportedChannelCount(dst_channels)) {
+        return SDL_SetError("Invalid destination channels");
+    }
+    if (src_rate <= 0) {
+        return SDL_SetError("Source rate is equal to or less than zero");
+    }
+    if (dst_rate <= 0) {
+        return SDL_SetError("Destination rate is equal to or less than zero");
+    }
+    if (src_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Source rate is too high");
+    }
+    if (dst_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Destination rate is too high");
+    }
+#endif
+
+    stream = SDL3_CreateAudioStream(src_format, src_channels, src_rate,
```

Comment:
We're creating a stream here that we don't use.

Reply:
I thought this would be useful as a sanity check.
should I remove it anyway ?

Reply:
It allocates memory and does a lot of sanity checks that we're doing here anyhow, so I'm inclined to say it should be removed.

Reply:
ok, 
so: pre-creating audio stream is removed.
and re-added the checks